CMakeLists.txt

```diff
@@ -71,10 +71,13 @@ option(BUILD_TESTING "Build the testing tree." OFF)
 include(CTest)
 add_custom_target(build_tests)
 set_target_properties(build_tests PROPERTIES EXCLUDE_FROM_ALL $<NOT:$<BOOL:${BUILD_TESTING}>>)
-add_subdirectory(tests/auth)
-add_subdirectory(tests/nativemessaging)
-add_subdirectory(tests/unit)
-add_subdirectory(tests/qml)
+
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten")
```

Comment:
Out of curiosity, does building for WASM set the `CROSS_COMPILING` variable? The best approach here would be to setup the tests to be built for the host system, and failing that we should just drop them when cross compiling (which will apply to iOS and Android too).